### ISSUE TITLE
What should be the CloudAMQP URL for Websockets?

### ISSUE THREAD

**Issue by ShuzhaoFeng** [cloudamqp/amqp-client.js#113]
Consider the following code snippet to connect to CloudAMQP:

```ts
  const cloud_amqp_url = "wss://rabbit-01.lmq.cloudamqp.com";
  
  const client = new AMQPWebSocketClient(
    cloud_amqp_url,
    "vhost & user",
    "password",
    "vhost & user"
  );

  client
    .connect()
    .then((connection) => {
      console.log("Connected to CloudAMQP");
      return connection.channel();
    })
    .then((channel) => {
      console.log("Channel created");
      channel.basicPublish("", "my-queue", "Hello, World!");
      console.log("Message sent");
    });
```

For `cloud_amqp_url`, the one starting with `amqps` does not work with the WebSocket Client. It would return an error message saying `Expected url scheme 'http' or 'https' but was 'amqps'`.

I went to the CloudAMQP and AMQP Client documentation then tried several things (`wss://rabbit-01.lmq.cloudamqp.com`, `wss://{User & Vhost}:{password}@rabbit.lmq.cloudamqp.com/{User & Vhost}`, `wss://rabbit-01.lmq.cloudamqp.com/wss`, `wss://rabbit-01.lmq.cloudamqp.com/wss/amqp`, `ws://rabbit-01.lmq.cloudamqp.com`), all of then either returns HTTP status 401 or 200, none matched with the expected 101.

What should be my URL for Websocket connection?

**Comment by dentarg** [comment]
The URL should look like this `wss://test-small-ivory-rat.rmq2.cloudamqp.com/ws/amqp` ([source](https://www.cloudamqp.com/docs/amqp_over_websockets.html))

However, it is not (yet) supported on LavinMQ plans (`lmq.cloudamqp.com`).

**Comment by ShuzhaoFeng** [comment]
@dentarg thanks for the reply.

I was aware of the documentation, but the example was too abstract (the example url doesn't have a username nor a password) for me to deduce what I need.

After your comment, I created a RabbitMQ instance with username `myuser:myuser` and password `XXXXXXXXXXXXXXX--XXXXXXXXXXXX`

I tried to connect both with 
```
wss://myuser:XXXXXXXXXXXXXXX--XXXXXXXXXXXX@codfish.rmq.cloudamqp.com/ws/amqp
```
and
```
wss://myuser:XXXXXXXXXXXXXXX--XXXXXXXXXXXX@codfish.rmq.cloudamqp.com/myuser/ws/amqp
```
neither of them seems to work as both would return a `404 Not Found` message.

Anything I missed here?
